Search, One-page forms and User mgmt section review (#10)
ezsystems/commerce-documentation: docs/guide/one_page_forms/implement_custom_one_page_form.md

# Implement custom one-page form

This example shows how to implement a form for ordering the product catalog in a printed form.

The user inputs their email address and confirms that they want to order the catalog.

> This sends as email to the shop administrator.

Reviewer: ```suggestion
This invokes an email to the shop administrator.
```

> This sends as email to the shop administrator.

Author: ```suggestion
An email is then sent to the shop administrator.
```

use Silversolutions\Bundle\EshopBundle\Entities\Forms\AbstractFormEntity;
use Symfony\Component\Validator\Constraints as Assert;
use Silversolutions\Bundle\EshopBundle\Entities\Forms\Constraints as SesAssert;

class OrderCatalog extends AbstractFormEntity
{
    /**
     * @Assert\NotBlank()
     * 
     * @var bool
     */
    protected $orderCatalog;

    /**
    * @Assert\NotBlank()
    * @SesAssert\Email() 
    *
    * @var string
    */
    protected $email;   

    /**
     * @return string
     */
    public function getOrderCatalog()
    {
        return $this->orderCatalog;
    }

    /**
     * @param string $orderCatalog
     */
    public function setOrderCatalog($orderCatalog)
    {
        $this->orderCatalog = $orderCatalog;
    }

    /**
    * @return string
    */
    public function getEmail()
    {
        return $this->email;
    }

    /**
    * @param string $email
    */
    public function setEmail($email)
    {
        $this->email = $email;
    }
}
```

### Form type

Form type defines which form attributes are rendered and how.
You can define other data here, such as labels, CSS classes, data attributes, etc.

> You should define this type as a service, so you can inject any logic that you need.

Reviewer: ```suggestion
You define this type as a service, so you can inject any logic that you need.
```

            ))  
            ->add('orderCatalog', 'checkbox', array(
                'required' => true,
                'label' => $this->transService->translate('orderCatalog')
            ))
        ;
    }

    /**
     * configure form options
     *
     * @param OptionsResolver $resolver
     * @return void
     *
     */
    public function configureOptions(OptionsResolver $resolver)
    {
        $resolver->setDefaults(array(
            'validation_groups' => function(FormInterface $form) {
                $groups = array();
                $groups[] = Constraint::DEFAULT_GROUP;
                return $groups;
            },
        ));
    }

    /**
     * Returns the name of this type.
     *
     * @return string The name of this type
     */
    public function getName()
    {
        return 'company_order_catalog_type';
    }
}
```

Service definition:

``` xml
<service id="company.order_catalog_type" class="Company\Bundle\ProjectBundle\Form\Type\OrderCatalogType">
    <argument type="service" id="silver_trans.translator" />    
</service>
```

### Form template

Then you need to prepare template that renders the form.

> See [How to render Symfony forms](https://symfony.com/doc/current/form/form_customization.html) for more information.

Reviewer: ```suggestion
See [How to render Symfony forms](https://symfony.com/doc/3.4/form/form_customization.html) for more information.
```
